maven-core/src/test/java/org/apache/maven/settings/PomConstructionWithSettingsTest.java

```diff
@@ -60,7 +60,7 @@ protected void customizeContainerConfiguration( ContainerConfiguration container
     {
         super.customizeContainerConfiguration( containerConfiguration );
         containerConfiguration.setAutoWiring( true );
-        containerConfiguration.setClassPathScanning( PlexusConstants.SCANNING_INDEX );
+        containerConfiguration.setClassPathScanning( PlexusConstants.SCANNING_ON );
```

Comment:
Why does this change? As far as I remember, Maven doesn't do classpath scanning, it relies on descriptors. @rfscholte correct me if I am wrong.

Reply:
This change is necessary for the JSR 330 classes to get picked up in `PlexusTestCase`. Maven runs after these changes (and `maven-integration-testing` passes), so it looks like something's picking them up without descriptors (something in [`MavenCli`](https://github.com/apache/maven/blob/edf61684c834090561954ea32d90d917e69c7b5e/maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java)?).

Reply:
Does it fail if you don't?

Reply:
Odd. Checking again, it passes without these changes.
It could have been an IDE issue; either way, I'll roll them back.